qemu/tests/migration_with_numa.py

```diff
@@ -0,0 +1,104 @@
+import re
+import logging
+
+from avocado.utils import process
+from virttest import env_process
+from virttest import error_context
+from virttest.utils_misc import get_mem_info
+from virttest.utils_misc import normalize_data_size
+from virttest.utils_misc import NumaInfo
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Bind guest node0 and node1 to 2 host nodes, do migration test
+
+    1. Boot src guest with 2 numa node and all bind to 2 host numa nodes
+    2. Migration
+    3. Check the memory in guest
+    4. Check the memory in dest host
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    def get_nodes_size(size_type='MemTotal', session=None):
+        """
+        Get the node size of each node in host/guest, sort with size.
+
+        :param size_type: the type of the node size
+        :param session: ShellSession object
+        """
+        numa_info = NumaInfo(session=session)
+        nodes_size = {}
+        numa_nodes = numa_info.online_nodes
+        for node in numa_nodes:
+            nodes_size[node] = float(numa_info.online_nodes_meminfo[node][size_type])
+        nodes_size = dict(sorted(nodes_size.items(), key=lambda item: item[1]))
+        return nodes_size
+
+    host_nodes_size = get_nodes_size(size_type='MemFree')
+    mem_devs = params.objects('mem_devs')
+    num = len(mem_devs)
+    for mem_dev in mem_devs:
+        size_mem = params.object_params(mem_dev).get('size_mem')
+        size_mem = float(normalize_data_size(size_mem, 'K'))
+        for node, size in host_nodes_size.items():
```

Comment:
Let's sort host_nodes_size by descending to avoid duplicate check.

Reply:
With function `get_nodes_size`, the return value is a sorted size, so the `host_nodes_size` already be sorted, and it is descending.

Reply:
Updated, thanks!